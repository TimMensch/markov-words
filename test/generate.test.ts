import assert from "assert";
import { generate, calculateStats as calculate } from "../index";

describe("Markov words", () => {
    describe("generate", () => {
        it("should generate aaa", () => {
            const result = generate(calculate(["aaa"]), 3);
            expect(result).toMatchSnapshot();
        });

        it("should generate aaa twice", () => {
            const result = generate(calculate(["aaa"]), 3, 2);
            expect(result).toMatchSnapshot();
        });

        it("should generate aaa and aab", () => {
            const stats = calculate(["aaa", "aab"]);
            let aaa = false;
            let aab = false;
            for (let i = 0; i < 100000; i++) {
                const result = generate(stats, 3);
                assert.equal(1, result.length);
                if (result[0] === "aaa") {
                    aaa = true;
                    if (aab) {
                        assert(true, "generated aaa and aab");
                        return;
                    }
                } else if (result[0] === "aab") {
                    aab = true;
                    if (aaa) {
                        assert(true, "generated aaa and aab");
                        return;
                    }
                } else {
                    fail(`${result[0]} Should be one of aaa or aab`);
                }
            }
        });

        it("should work with a large array of words", () => {
            const words = [
                "ability",
                "analyst",
                "artificial",
                "bout",
                "challenge",
                "conception",
                "crooked",
                "demand",
                "deny",
                "discussion",
                "dispute",
                "dodge",
                "double",
                "doubtful",
                "education",
                "enigma",
                "exploration",
                "fancy",
                "have",
                "head",
                "impossible",
                "inquiry",
                "inquisitive",
                "intellectual",
                "intriguing",
                "investigate",
                "judgment",
                "know",
                "knowing",
                "known",
                "learning",
                "light",
                "maneuver",
                "mark",
                "mind",
                "nose",
                "notice",
                "observation",
                "pay",
                "plea",
                "plot",
                "polite",
                "possess",
                "problem",
                "profound",
                "prying",
                "pump",
                "pursue",
                "query",
                "questioning",
                "reading",
                "reason",
                "rebut",
                "request",
                "retraction",
                "scent",
                "school",
                "shift",
                "sifting",
                "solution",
                "study",
                "surprise",
                "take",
                "theory",
                "trial",
                "trick",
                "wit",
            ];

            const stats = calculate(words);
            const result = generate(stats, 5, 20);

            assert(result != null, "Should be able to generate");
        });
    });
});
