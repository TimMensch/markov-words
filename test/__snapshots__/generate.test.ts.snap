// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markov words generate should generate aaa 1`] = `
Array [
  "aaa",
]
`;

exports[`Markov words generate should generate aaa twice 1`] = `
Array [
  "aaa",
  "aaa",
]
`;
